src/main/java/org/folio/innreach/domain/processor/PatronHoldInnReachCirculationProcessor.java

```diff
@@ -38,11 +45,17 @@ public InnReachResponseDTO process(String trackingId, String centralCode, Transa
 
     if (innReachTransaction.isPresent()) {
       log.info("Transaction patron hold with trackingId [{}] and centralCode [{}] exists, start to update...", trackingId, centralCode);
+
       updateTransactionPatronHold((TransactionPatronHold) innReachTransaction.get().getHold(), transactionHold);
+
+      patronHoldService.updateVirtualItems(innReachTransaction.get());
     } else {
       log.info("Transaction patron hold with trackingId [{}] and centralCode [{}] doesn't exist, create a new one...", trackingId, centralCode);
+
       InnReachTransaction newTransactionWithPatronHold = createTransactionWithPatronHold(trackingId, centralCode, transactionHold);
-      transactionRepository.save(newTransactionWithPatronHold);
+      var transaction = transactionRepository.save(newTransactionWithPatronHold);
+
+      patronHoldService.createVirtualItems(transaction);
```

Comment:
According to the method definition, this operation is asynchronous, meaning patron hold transaction will be created and successful response returned to the client. 
What should happen with the transaction is async method fails at some step, e.g item is not created: 
- leave everything as is?
- delete the transaction?
- other option

Reply:
well, the requirements indeed only describe the positive scenario, which doesn't mean we shouldn't think about negative cases. Otherwise we might face with serious problems when the code is on prod.
I suggest to at least ask PO about those potential negative scenarios and let him to decide whether it's a show stopper for the story or not